### PR TITLE
use refs to focus on first error in application before going to next

### LINKED ISSUE #289: Hacker App > Validation Updates
When validation unmet -> the window should scroll into view the question that caused the error (and the pop-up should not exist)

### PR DESCRIPTION
**Pull request by andykwan1124**
## Description
focusError

Closes #289 

Trying to go to the next page of the application while there are errors on the current page will cause the first error on the page to be focused.  This is done for the mandatory questions.

### CODE REVIEW

**Comment by andykwan1124** [comment]
I uploaded a rather long gif... Might take some time to load for u guys.

File: src/containers/Application/Part1.js [hunk 1]
```diff
@@ -43,7 +75,12 @@ export default () => {
     if (href === '/application/part-2') {
       const newErrors = validate(application.basicInfo)
       if (checkForError(newErrors)) {
-        window.alert('Please complete all required fields.')
+        for (let question of questionsByOrder) {
+          if (newErrors[question]) {
+            refs[`${question}Ref`].current.focus()
```
Comment (panjenny0) [inline src/containers/Application/Part1.js:80]: wow smart

**Review by panjenny0 — APPROVED** [review]
wow this is great 💯 🚢 
appreciate you adding the gif too - was easy to see how it works

**Review by kevin-zou — APPROVED** [review]
This is some wrinkly brain energy right here 🧠

**Pull request merged**